src/box/box.cc

```diff
@@ -3555,23 +3589,47 @@ bootstrap(const struct tt_uuid *instance_uuid,
 	 * with connecting to 'replication_connect_quorum' masters.
 	 * If this also fails, throw an error.
 	 */
-	box_restart_replication();
-
-	struct replica *master = replicaset_find_join_master();
-	assert(master == NULL || master->applier != NULL);
-
-	if (master != NULL && !tt_uuid_is_equal(&master->uuid, &INSTANCE_UUID)) {
-		bootstrap_from_master(master);
-		/* Check replica set UUID */
-		if (!tt_uuid_is_nil(replicaset_uuid) &&
-		    !tt_uuid_is_equal(replicaset_uuid, &REPLICASET_UUID)) {
-			tnt_raise(ClientError, ER_REPLICASET_UUID_MISMATCH,
-				  tt_uuid_str(replicaset_uuid),
-				  tt_uuid_str(&REPLICASET_UUID));
+
+	struct replica *master;
+
```

Comment:
Please, reduce number of empty lines in the code. Usually it is fine here and there, but tbh in this patch it seems too much.

Reply:
I can't reduce number of empty lines in the code. Number of empty lines equal to 8 default in this file. I created the new function `check_replicaset_uuid()`

Reply:
I am quite sure the file has many more than just 8 empty lines. But there seems to be misunderstanding. I meant "empty lines", not "indentation level".

Reply:
The new function has a bit unclear name. "Check uuid" for what? For being invalid? For not matching our own `REPLICASET_UUID`? Besides, it is used in a single place anyway. There are ways to reduce indentation in such cases without factoring any code out:
```Diff
diff --git a/src/box/box.cc b/src/box/box.cc
index 53252b768..6d61c8d2c 100644
--- a/src/box/box.cc
+++ b/src/box/box.cc
@@ -3550,20 +3550,6 @@ bootstrap_from_master(struct replica *master)
 	return true;
 }
 
-/*
- * Check replica set UUID
- */
-static void
-check_replicaset_uuid(const struct tt_uuid *replicaset_uuid)
-{
-	if (!tt_uuid_is_nil(replicaset_uuid) &&
-	    !tt_uuid_is_equal(replicaset_uuid, &REPLICASET_UUID)) {
-		tnt_raise(ClientError, ER_REPLICASET_UUID_MISMATCH,
-			  tt_uuid_str(replicaset_uuid),
-			  tt_uuid_str(&REPLICASET_UUID));
-	}
-}
-
 /**
  * Bootstrap a new instance either as the first master in a
  * replica set or as a replica of an existing master.
@@ -3620,19 +3606,24 @@ bootstrap(const struct tt_uuid *instance_uuid,
 	while (true) {
 		box_restart_replication();
 		master = replicaset_find_join_master();
-
-		if (master != NULL &&
-		    !tt_uuid_is_equal(&master->uuid, &INSTANCE_UUID)) {
-			if (bootstrap_from_master(master)) {
-				check_replicaset_uuid(replicaset_uuid);
-				break;
-			}
-		} else {
+		if (master == NULL ||
+		    tt_uuid_is_equal(&master->uuid, &INSTANCE_UUID)) {
 			bootstrap_master(replicaset_uuid);
 			*is_bootstrap_leader = true;
 			break;
 		}
 
+		bool is_bootstrapped = bootstrap_from_master(master);
+		if (is_bootstrapped && !tt_uuid_is_nil(replicaset_uuid) &&
+		    !tt_uuid_is_equal(replicaset_uuid, &REPLICASET_UUID)) {
+			tnt_raise(ClientError, ER_REPLICASET_UUID_MISMATCH,
+				  tt_uuid_str(replicaset_uuid),
+				  tt_uuid_str(&REPLICASET_UUID));
+		}
+		if (is_bootstrapped) {
+			*is_bootstrap_leader = false;
+			break;
+		}
 		if (!say_once) {
 			say_info("rebootstrap failed, will retry "
 				 "every %.2lf second", timeout);
```

Reply:
Yea, I misunderstood you. Fixed.